narwhals/_pandas_like/dataframe.py

```diff
@@ -277,6 +278,35 @@ def join(
         if isinstance(right_on, str):
             right_on = [right_on]
 
+        if how == "cross":
+            if self._implementation in {"modin", "cudf"} or (
+                self._implementation == "pandas"
+                and (pd := get_pandas()) is not None
+                and parse_version(pd.__version__) < parse_version("1.2.0")
+            ):
+                import secrets
+
+                n_bytes = max(len(c.encode("utf-8")) for c in self.columns) + 1
```

Comment:
The +1 should guarantee that it's not already in the columns

Reply:
mmmh wondering if we can have issues as pandas column names do not need to be strings